### DISCUSSION TITLE
pico over USB not connecting to Android device

### DISCUSSION THREAD

**Discussion by Devnol** [fhdm-dev/scoppy#81] [Q&A]
Hello, I am trying to connect my pico to my Galaxy A12 but when I plug it in nothing happens. I'm using a generic USB-C to micro-B adapter and then a micro-B to A OTG dongle, which connects to a USB-A to micro-B cable, which attaches to the Pico. The onboard LED does blink and the cable does work when I try a flash drive or some other USB device, but I get no prompts on Scoppy when I connect the Pico.

**Comment by fhdm-dev** [comment]
Hi,

Is the connection type in the app set to USB? You can see the currently selected connection type in the connection badge at the lower left corner of the app. If it's set to either Wi-Fi or Demo then a USB connection won't be established. If you need to change the connection type then tap the badge and the option will be in the menu that pops up.

If that is not the problem then please connect the Pico to your computer in boolsel mode and post the output from the _picotool info -a_ command.

A photo might also be useful to help me troubleshoot the issue.

.... and restarting your phone might help ....

Thanks,
Brett

**Reply by Devnol** [reply]
Hello, sorry for the late reply, couldn't figure out why picotool wasn't compiling (turns out it was trying to use the devkitpro pkg-config 🤦🏼). Yes, I have selected USB in the app and I did try restarting the phone to no avail. 

Here is the output of the picotool info command you asked for:
```
Program Information
 name:          scoppy-pico-v11
 version:       11
 web site:      https://github.com/fhdm-dev/scoppy
 description:   build config: default 1
 features:      SMPS-PWM
                double reset -> BOOTSEL
                UART stdin / stdout
 binary start:  0x10000000
 binary end:    0x1000dc38

Fixed Pin Information
 0:   UART0 TX
 1:   UART0 RX
 2:   Voltage Range CH1 Bit1
 3:   Voltage Range CH1 Bit0
 4:   Voltage Range CH2 Bit1
 5:   Voltage Range CH2 Bit0
 6:   Logic Analyser First Channel
 13:  Logic Analyser Last Channel
 14:  WiFi Status LED
 15:  Trigger LED
 22:  Signal Generator 1
 23:  SMPS_GPIO
 25:  Status LED
 26:  CH1 ADC
 27:  CH2 ADC

Build Information
 sdk version:       1.4.0
 pico_board:        pico
 boot2_name:        boot2_w25q080
 build date:        Sep  8 2022
 build attributes:  Release

Device Information
 flash size:   2048K
 ROM version:  2
 ```
 The connection I'm using is as follows (white microUSB to type C adapter goes to my phone):
 
 
 Also when the Pico is plugged in Android says "charging connected device" and does not let me switch master/slave of the USB connection:

**Reply by Devnol** [reply]
(gh mobile won't let me edit the above message so here are the attached images):

![20221018_203545.jpg](https://user-images.githubusercontent.com/58105634/196504636-681e3de3-362c-4434-a42f-58fe6220deb0.jpg)

![Screenshot_20221018-203454_Settings.png](https://user-images.githubusercontent.com/58105634/196504673-dcebb7f2-c0f5-4451-9756-d10e4d2e0432.png)

**Reply by fhdm-dev** [reply]
Thank for the info. That all looks fine (except not being able to switch USB settings to controlled by 'This device' of course).

You say a flash drive works - I assume you're using the USB-C to micro-B adapter and then the micro-B to A OTG dongle plugged into the drive. If so, that just leaves the USB-A to micro-B cable as a possible culprit. Any chance that cable is faulty or is a charge-only cable?

Another thing that might be worth looking at. If you tap the connection badge and then _Connected Device_ you'll see an option to display the Android USB Info. Does that show any connected devices?

**Reply by Devnol** [reply]
It can't be the cable either, I used that same cable to run picotool and flash the uf2 in the first place. If I go to connected devices nothing shows up either
